addons/test_mail/tests/test_message_post.py

```diff
@@ -225,115 +260,307 @@ def test_notify_recipients_internals(self):
+
+    @users('employee')
+    def test_message_post_inactive_follower(self):
+        """ Test posting with inactive followers does not notify them (e.g. odoobot) """
+        test_record = self.env['mail.test.simple'].browse(self.test_record.ids)
+        test_record._message_subscribe(self.user_employee_2.partner_id.ids)
+        self.user_employee_2.write({'active': False})
+        self.partner_employee_2.write({'active': False})
+
+        with self.assertPostNotifications([{'content': 'Test', 'notif': []}]):
+            test_record.message_post(
+                body='Test',
+                message_type='comment',
+                subtype_xmlid='mail.mt_comment')
 
     @mute_logger('odoo.addons.mail.models.mail_mail')
-    def test_post_notifications_keep_emails(self):
-        self.test_record.message_subscribe(partner_ids=[self.user_admin.partner_id.id])
-
-        msg = self.test_record.with_user(self.user_employee).message_post(
-            body='Test', subject='Test',
-            message_type='comment', subtype_xmlid='mail.mt_comment',
-            partner_ids=[self.partner_1.id, self.partner_2.id],
-            mail_auto_delete=False
-        )
+    @users('employee')
+    def test_message_post_keep_emails(self):
+        test_record = self.env['mail.test.simple'].browse(self.test_record.ids)
+        test_record.message_subscribe(partner_ids=self.partner_employee_2.ids)
+
+        with self.mock_mail_gateway(mail_unlink_sent=True):
+            msg = test_record.message_post(
+                body='Test', subject='Test',
+                message_type='comment', subtype_xmlid='mail.mt_comment',
+                partner_ids=[self.partner_1.id, self.partner_2.id],
+                mail_auto_delete=False
+            )
 
         # notifications emails should not have been deleted: one for customers, one for user
         self.assertEqual(len(self.env['mail.mail'].sudo().search([('mail_message_id', '=', msg.id)])), 2)
 
     @mute_logger('odoo.addons.mail.models.mail_mail')
-    def test_post_notifications_emails_tweak(self):
+    def test_message_post_notifications_emails_tweak(self):
         pass
         # we should check _notification_groups behavior, for emails and buttons
 
     @mute_logger('odoo.addons.mail.models.mail_mail')
-    def test_post_attachments(self):
+    @users('employee')
+    def test_message_post_w_attachments(self):
         _attachments = [
             ('List1', b'My first attachment'),
             ('List2', b'My second attachment')
         ]
-        _attach_1 = self.env['ir.attachment'].with_user(self.user_employee).create({
+        _attach_1 = self.env['ir.attachment'].create({
             'name': 'Attach1',
             'datas': 'bWlncmF0aW9uIHRlc3Q=',
-            'res_model': 'mail.compose.message', 'res_id': 0})
-        _attach_2 = self.env['ir.attachment'].with_user(self.user_employee).create({
+            'res_model': 'mail.compose.message',
+            'res_id': 0
```

Comment:
(check also other occurrence in this pr)
```suggestion
            'res_id': 0,
```